cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -428,3 +428,84 @@ func (s *Server) GetInstalledPackageDetail(ctx context.Context, request *corev1.
 	}
 	return response, nil
 }
+
+// CreateInstalledPackage creates an installed package managed by the 'kapp_controller' plugin
+func (s *Server) CreateInstalledPackage(ctx context.Context, request *corev1.CreateInstalledPackageRequest) (*corev1.CreateInstalledPackageResponse, error) {
+	log.Infof("+kapp-controller CreateInstalledPackage")
+
+	// Validate the request
+	if request.GetAvailablePackageRef().GetContext().GetNamespace() == "" || request.GetAvailablePackageRef().GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "required context or identifier not provided")
+	}
+	if request == nil || request.GetAvailablePackageRef() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no request AvailablePackageRef provided")
+	}
+	if request.GetName() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no request Name provided")
+	}
+	if request.GetTargetContext() == nil || request.GetTargetContext().GetNamespace() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no request TargetContext namespace provided")
+	}
+
+	// Retrieve the proper parameters from the request
+	packageRef := request.GetAvailablePackageRef()
+	cluster := packageRef.GetContext().GetCluster()
+	reconciliationOptions := request.GetReconciliationOptions()
+	pkgVersion := request.GetPkgVersionReference().GetVersion()
+	installedPackageName := request.GetName()
+	values := request.GetValues()
+	targetNamespace := request.GetTargetContext().GetNamespace()
```

Comment:
Why not pull out the target cluster as well? Currently it looks like it's just assuming the available package cluster. AFAICT, we don't lose anything by assuming multi-cluster support here, rather than explicitly not supporting it?